migration/migration.go

```diff
@@ -620,23 +621,24 @@ func createOrUpdateSpace(ctx context.Context, spaceRepo *space.GormRepository, i
 }
 
 func createSpace(ctx context.Context, spaceRepo *space.GormRepository, id uuid.UUID, description string) error {
-	_, err := spaceRepo.Load(ctx, id)
-	cause := errs.Cause(err)
-	newSpace := &space.Space{
-		Description: description,
-		Name:        "system.space",
-		ID:          id,
-	}
-	switch cause.(type) {
-	case errors.NotFoundError:
+	err := spaceRepo.CheckExists(ctx, id.String())
+	switch reflect.TypeOf(err) {
+	case reflect.TypeOf(&goa.ErrorResponse{}):
```

Comment:
I still don't get why we use the cause function to check which error it was. Also goa.ErrorResponse is a bad idea to use because it introduces goa to our repository layer, right? I'd like to keep goa out from there. If I'm wrong about that introduction, please let me know. @aslakknutsen any thoughts?